src/modules/location/index.ts

```diff
@@ -320,37 +320,45 @@ export class LocationModule {
    * Returns a random [ISO_3166-1](https://en.wikipedia.org/wiki/ISO_3166-1) country code.
    *
    * @param options The code to return or an options object. Defaults to `{}`.
-   * @param options.variant The variant to return. Can be either `'alpha-2'` (two-letter code)
-   * or `'alpha-3'` (three-letter code). Defaults to `'alpha-2'`.
+   * @param options.variant The variant to return. Can be either `'alpha-2'` (two-letter code),
+   * `'alpha-3'` (three-letter code) or `'numeric'` (numeric code). Defaults to `'alpha-2'`.
    *
    * @example
    * faker.location.countryCode() // 'SJ'
    * faker.location.countryCode('alpha-2') // 'GA'
    * faker.location.countryCode('alpha-3') // 'TJK'
+   * faker.location.countryCode('numeric') // '528'
    *
    * @since 8.0.0
    */
   countryCode(
     options:
       | 'alpha-2'
       | 'alpha-3'
+      | 'numeric'
       | {
           /**
            * The code to return.
-           * Can be either `'alpha-2'` (two-letter code)
-           * or `'alpha-3'` (three-letter code).
+           * Can be either `'alpha-2'` (two-letter code),
+           * `'alpha-3'` (three-letter code)
+           * or `'numeric'` (numeric code).
            *
            * @default 'alpha-2'
            */
-          variant?: 'alpha-2' | 'alpha-3';
+          variant?: 'alpha-2' | 'alpha-3' | 'numeric';
         } = {}
   ): string {
     if (typeof options === 'string') {
       options = { variant: options };
     }
 
     const { variant = 'alpha-2' } = options;
-    const key = variant === 'alpha-3' ? 'alpha3' : 'alpha2';
+    const key =
+      variant === 'alpha-3'
+        ? 'alpha3'
+        : variant === 'numeric'
+        ? 'numeric'
+        : 'alpha2';
```

Comment:
Might be me, but having an IIFE switch here would improve the readability:

 ```suggestion
    const key = (() => {
      switch(variant) {
        case 'numeric': 
          return 'numeric';
        case 'alpha-3': 
          return 'alpha3';
        case 'alpha-2': 
        default: 
          return 'alpha2';
      }
    })();
```

Absolute best would be to extract the IIFE into a function over the class (we are not able to use private functions) and use it. That way we could apply a nice name additionally.

Reply:
It is only used in a single place. Not sure whether we need an external method for that

Reply:
The switch version is nicer. Nested ternaries are very hard to read.

Reply:
A more easier solution would be:

```suggestion
    const key = {
      'alpha-3': 'alpha3',
      'numeric': 'numeric',
    }[variant] ?? 'alpha2';
```